# Copy this patch file to feeds/packages/net/openvpn/patches
# Will be executed automatically
diff -u -r -x .DS_Store openvpn-2.6.4.old/src/openvpn/forward.c openvpn-2.6.4.new/src/openvpn/forward.c
--- a/src/openvpn/forward.c
+++ b/src/openvpn/forward.c
@@ -928,7 +928,10 @@
 
     status = link_socket_read(c->c2.link_socket,
                               &c->c2.buf,
-                              &c->c2.from);
+                              &c->c2.from,
+                              c->options.ce.xormethod,
+                              c->options.ce.xormask,
+                              c->options.ce.xormasklen);
 
     if (socket_connection_reset(c->c2.link_socket, status))
     {
@@ -1782,7 +1785,10 @@
                 socks_preprocess_outgoing_link(c, &to_addr, &size_delta);
 
                 /* Send packet */
-                size = link_socket_write(c->c2.link_socket, &c->c2.to_link, to_addr);
+                size = link_socket_write(c->c2.link_socket, &c->c2.to_link, to_addr,
+                                         c->options.ce.xormethod,
+                                         c->options.ce.xormask,
+                                         c->options.ce.xormasklen);
 
                 /* Undo effect of prepend */
                 link_socket_write_post_size_adjust(&size, size_delta, &c->c2.to_link);
